### DISCUSSION TITLE
How to enable media encryption on WSS?

### DISCUSSION THREAD

**Discussion by SergSV12** [juha-h/baresip-studio#433] [Q&A]
Help setting up WSS in baresip+ with media encryption. How to enable SRTP protocol?
What I do: is create an account, add “sip:siptest.test.net:8089;transport=wss” to “Outgoing proxies”, select “SRTP-MANDF” for media encryption.
Registration is successful. The call to baresip+ is working. The video conversation is established. But baresip+ sends unencrypted media data. How to fix it?

**Comment by juha-h** [comment]
Please see the Wiki about Debugging info.  WebRTC uses DTLS-SRTPF for encryption.

**Comment by SergSV12** [comment]
I see an error in the PBX log:
"video Handshake failure 1. This may happen when you use legacy DTLS v1.0 (legacyDTLS channel var is set) but endpoint requires DTLS v1.2."

How to disable DTLS v1.0 and force enable DTLS v1.2? 
![image](https://github.com/juha-h/baresip-studio/assets/72083838/a6b42cd3-f44c-4034-8a88-16e56428a03a)

**Comment by juha-h** [comment]
Looks to me that you need to fix the issue in your PBX, i.e., make it support DTLS v1.2.

**Comment by SergSV12** [comment]
Some calls work fine, but some fail. I compared the packages in Wireshark. The packages from PBX are no different. Baresip's reaction varies.
In some cases it sends an "Alert".
What could be the reason? How to fix this problem?

Is working:
![image](https://github.com/juha-h/baresip-studio/assets/72083838/0a5250a6-8e50-47df-823b-638a3ca07ead)

Is don't working:
![image](https://github.com/juha-h/baresip-studio/assets/72083838/4176bf3e-0ad2-48ed-aa39-3fa0e55fb960)

I am attaching network dumps:
[pcap.zip](https://github.com/juha-h/baresip-studio/files/13965261/pcap.zip)

Logcat:
![image](https://github.com/juha-h/baresip-studio/assets/72083838/0200e903-9ba5-445e-89c5-59bb9c9c6ea2)

**Comment by juha-h** [comment]
I'm not a crypto expert.  Try asking some crypto forum or search from the
net what could cause such TLS error messages.  baresip is using OpenSSL
version 3.1 from March 14, 2023.

**Comment by juha-h** [comment]
I made several tests between baresip+ and LInphone using DTLS-SRTPF media encryption and in all tests DTLSv1.2 negotiation worked without errors.  So I suspect that there is something wrong with your PBX.  In any case, there is nothing I can do to help.

**Comment by SergSV12** [comment]
What PBX did you use? What is the version of OpenSSL on the PBX?
